acceptance: add six node cluster in steady state

tbg opened the pull request:
This steady-state six node cluster is a good starting point for tests
which downreplicate or attempt to delete large amounts of data. The
relevant contribution here is having created the data file.

Note that 1038 is the number of Replicas per Store taken from the
admin UI. 56GB is the size of the data directory, though the first
node clocks in at 112GB, which is a bit suspicious and might warrant
an investigation (a GCed replica should have its on-disk state removed;
perhaps RocksDB compactions can take care of the rest? cc @petermattis).

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7944)

<!-- Reviewable:end -->

cuongdo commented:
I agree that the data set is valuable. Does this test have value in and of itself? Maybe in conjunction with `tf.keep-cluster`?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-:-KN8IzxyDYtL5E2NT7ww:-866405723)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
I'll certainly use it for setting up clusters. But added value would be
dropping the table or running schema changes on it. I don't want to pursue
that right now, hence this basic PR.

On Wed, Jul 20, 2016, 19:19 Cuong Do <notifications@github.com> wrote:

> I agree that the data set is valuable. Does this test have value in and of
> itself? Maybe in conjunction with tf.keep-cluster?
> ------------------------------
>
> Review status: 0 of 1 files reviewed at latest revision, all discussions
> resolved, all commit checks successful.
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-:-KN8IzxyDYtL5E2NT7ww:-866405723>*
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/7944#issuecomment-234017601>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE135AsvmUoJUgbOQNa3CgbOEFlfDDGYks5qXlhigaJpZM4JQ9WX>
> .
>
-- 

-- Tobias

cuongdo commented:
<img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

OK, makes sense. Just wanted to make sure I wasn't missing something. Maybe add a short comment to that effect so that someone browsing the code understands the intent.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-:-KN8M0V8dPC1rykW97Ws:2044434018)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: add six node cluster in steady state

This steady-state six node cluster is a good starting point for tests
which downreplicate or attempt to delete large amounts of data. The
relevant contribution here is having created the data file.

Note that 1038 is the number of Replicas per Store taken from the
admin UI. 56GB is the size of the data directory, though the first
node clocks in at 112GB, which is a bit suspicious and might warrant
an investigation (a GCed replica should have its on-disk state removed;
perhaps RocksDB compactions can take care of the rest?).

tbg committed:
acceptance: less interaction when destroying cluster

tschottdorf commented:
PTAL, I added a bit of flag refactoring. cc @jordanlewis as this might require an adjustment to the nightlies.

tbg committed:
acceptance: consolidate keep-cluster flags

cuongdo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> w/ comment & question

---

Reviewed 1 of 1 files at r1.
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[acceptance/util_test.go, line 300 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-KNCvT6OS86qr1ZgAQDh:-KNCvT6OS86qr1ZgAQDi:1253969762) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb6901a598c578bd35ee65cc34d2c0cf9ffb8323/acceptance/util_test.go#L300)):*
> ```Go
> 	}
> 	if err := f.WaitReady(5 * time.Minute); err != nil {
> 		_ = f.Destroy(t)
> ```

not new to this PR, but should we log the error instead of suppressing it?

---

*[acceptance/terrafarm/farmer.go, line 167 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-KNCuyvPIBiU58CydJX0:-KNCuyvPIBiU58CydJX1:1858058823) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb6901a598c578bd35ee65cc34d2c0cf9ffb8323/acceptance/terrafarm/farmer.go#L167)):*
> ```Go
> 	f.CollectLogs()
> 	if f.LogDir != "" {
> 		defer f.logf("logs copied to %s\n", f.AbsLogDir())
> ```

not specific to this PR, but why do we have `(*Farmer).logf` instead of using `util.Log`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-:-KNCwE9EbufnbkeTHTTs:1163764090)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: log an error

tschottdorf commented:
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[acceptance/util_test.go, line 300 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-KNCvT6OS86qr1ZgAQDh:-KND3RRSD4JnUXfKnnqB:926693478) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb6901a598c578bd35ee65cc34d2c0cf9ffb8323/acceptance/util_test.go#L300)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> not new to this PR, but should we log the error instead of suppressing it?
</details>
Good point, done.

---

*[acceptance/terrafarm/farmer.go, line 167 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944#-KNCuyvPIBiU58CydJX0:-KND2iGWdWd9Vhj_wxjP:99801916) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb6901a598c578bd35ee65cc34d2c0cf9ffb8323/acceptance/terrafarm/farmer.go#L167)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> not specific to this PR, but why do we have `(*Farmer).logf` instead of using `util.Log`?
</details>
`logf` writes to an output stream associated to the farmer (I think it's currently always `os.Stderr`, though). The log header is pretty useless for these kinds of log messages (which I think is most of the reason).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7944)*
<!-- Sent from Reviewable.io -->
